utils/logger/logger_test.go

```diff
@@ -0,0 +1,348 @@
+// this is a new logger interface for mattermost
+
+package logger
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+type LogMessage struct {
+	Context map[string]string
+	File    string
+	Message string
+}
+
+// ensures that values can be recorded on a Context object, and that the data in question is serialized as a part of the log message
+func TestSerializeContext(t *testing.T) {
+	t.Run("Context values test", func(t *testing.T) {
+		ctx := context.Background()
+
+		expectedUserID := "some-fake-user-id"
+		ctx = WithUserID(ctx, expectedUserID)
+
+		expectedRequestID := "some-fake-request-id"
+		ctx = WithRequestID(ctx, expectedRequestID)
+
+		serialized := serializeContext(ctx)
+
+		userID, ok := serialized["user_id"]
+		if !ok {
+			t.Error("UserID was not serialized")
+		}
+		if userID != expectedUserID {
+			t.Errorf("UserID = %v, want %v", userID, expectedUserID)
+		}
+
+		requestID, ok := serialized["request_id"]
+		if !ok {
+			t.Error("RequestID was not serialized")
+		}
+		if requestID != expectedRequestID {
+			t.Errorf("RequestID = %v, want %v", requestID, expectedRequestID)
+		}
```

Comment:
Fixed the ID/Id thing, although it's in contravention of official Google style, and a subject of much debate in the [golang/lint project](https://github.com/golang/lint/issues/89). 😄